erpnext/accounts/doctype/share_transfer/share_transfer.js

```diff
@@ -26,10 +31,74 @@ frappe.ui.form.on('Share Transfer', {
 		if (frm.doc.no_of_shares != undefined || frm.doc.no_of_shares != null){
 			erpnext.share_transfer.update_amount(frm);
 		}
+	},
+	company: async function(frm) {
+		if (frm.doc.company) {
+			let currency = (await frappe.db.get_value("Company", frm.doc.company, "default_currency")).message.default_currency;
+			frm.set_query("equity_or_liability_account", function() {
+				return {
+					filters: {
+						"is_group":0,
+						"root_type": ["in",["Equity","Liability"]],
+						"company": frm.doc.company,
+						"account_currency": currency
+					}
+				};
+			});
+			frm.set_query("asset_account", function() {
+				return {
+					filters: {
+						"is_group":0,
+						"root_type":"Asset",
+						"company": frm.doc.company,
+						"account_currency": currency
+					}
+				};
+			});
+		}
 	}
 });
 
 erpnext.share_transfer.update_amount = function(frm) {
 	frm.doc.amount = frm.doc.no_of_shares * frm.doc.rate;
 	frm.refresh_field("amount");
 };
+
+erpnext.share_transfer.make_jv = function (frm) {
+	var account,payment_account,credit_applicant_type,credit_applicant,
+		debit_applicant_type,debit_applicant;
+
+	if (frm.doc.transfer_type == "Transfer") {
+		account = frm.doc.equity_or_liability_account;
+		payment_account = frm.doc.equity_or_liability_account;
+		credit_applicant_type = "Shareholder";
+		credit_applicant = frm.doc.to_shareholder;
+		debit_applicant_type = "Shareholder";
+		debit_applicant = frm.doc.from_shareholder;
+	}
+	else {
+		account =(frm.doc.transfer_type == "Issue") ? frm.doc.asset_account : frm.doc.equity_or_liability_account;
```

Comment:
As this conditional statement `(frm.doc.transfer_type == "Issue")` is repeated 6 times, it will be always better writing a `if` condition instead for better readability.